src/commons/navigationBar/subcomponents/SicpNavigationBar.tsx

```diff
@@ -222,212 +111,541 @@ const SicpNavigationBar: React.FC = () => {
+    searchInput: string,
+    isSubMenueHidden: (searchInput: string) => Boolean,
+    getResults: () => any[],
+    buildMenuEntry: (searchResult: any, index: number) => React.ReactNode,
+    buildMenuWith: (
+      children: React.ReactNode,
+      ref: React.RefObject<HTMLDivElement>
+    ) => React.ReactNode
+  ) => {
+    if (isSubMenueHidden(searchInput)) {
+      return <></>;
+    }
+    const children = getResults().map(buildMenuEntry);
+    return buildMenuWith(children, autocompleteMenuRef);
   };
 
-  const userSearch = (
-    <div className="userSearch" style={{ position: 'relative' }} key="userSearch">
-      <div className="userSearch-inner">
-        <div style={{ display: 'inline-flex' }}>
-          <InputGroup
-            placeholder="Search"
-            value={searchQuery}
-            onChange={event => handleUserSearchChange(event.target.value)}
-          />
-          <ControlButton
-            label="Text"
-            icon={IconNames.SEARCH}
-            onClick={() => handleSearchButton()}
-          />
-        </div>
+  const userSearchResultSubMenu = (searchInput: string) => {
+    const focusResult = (result: string) => {
+      result = result.replaceAll('\n', ' ').toLowerCase();
+      const startIndex = result.indexOf(searchInput);
+      let start = startIndex;
+      while (start > 0) {
+        if (result[start - 1].match(/[^a-zA-Z, _]/)) {
+          console.log('break at ' + (start - 1) + ' ' + result[start - 1]);
+          break;
+        }
+        start--;
+      }
+      const endIndex = startIndex + searchInput.length;
+      let end = endIndex;
+      while (end < result.length) {
+        if (result[end].match(/[^a-zA-Z _,]/)) {
+          console.log('break at ' + end + ' ' + result[end]);
+          break;
+        }
+        end++;
+      }
+      let subStr = result.slice(start, end);
+      console.log('subStr is ' + subStr);
+      if (start > 0) {
+        subStr = '...' + subStr;
+      }
+      if (end < result.length) {
+        subStr = subStr + '...';
+      }
+      return subStr;
+    };
+
+    const isUserSearchSubMenuHidden = (searchInput: string) => {
+      return !(isSubmenuVisible === searchInput);
+    };
+
+    const getUserSearchResults = () => {
+      return results;
+    };
+    const getDisplayedIndex = (id: string) => {
+      const index = id.indexOf('#');
+      const numId = index === -1 ? id : id.slice(0, index);
+      return numId + ': ';
+    };
+
+    const buildUserSearchResultsMenuEntry = (result: any, index: number) => (
+      <div
+        style={{
+          margin: '0',
+          width: '100%',
+          backgroundColor: focusedSearchResultIndex !== index ? 'white' : 'grey',
+          border: '1px solid black'
+        }}
+        key={index}
+        onClick={() => {
+          setIndexAutoCompleteCouldShow(false);
+          setSearchAutoCompleteCouldShow(false);
+          window.location.href = `http://localhost:8000/sicpjs/${result}`;
```

Comment:
You might have missed out on this, it should be the current URL of the page (instead of `http://localhost:8000`, which means you don't actually need the protocol + host parts of the URL)